site/content/blog/2025-02-17-portswigger-lab-walkthrough-broken-brute-force-protection-ip-block/index.md

```diff
@@ -0,0 +1,793 @@
+
+> [!NOTE]
+> This bypass — taking advantage of a logic flaw to reset the failure counter —  enables us to brute-force Carlos’ password and successfully solve this lab. Although not the point of the lab, we can also wait out the IP block since we’re dealing with only a few credentials. This is covered in “Method 2: Wait Out the IP Block”.
+
+What about changing our IP address? I used the “X-Forwarded-For” header successfully in another lab. But the server here appears to block IPs based on the client's actual IP address (determined by the network layer rather than HTTP headers from the application layer), which makes sense in the context of rate-limiting. Rotating IP addresses via a proxy or VPN could work, but we won’t explore that here.
+
+## Brute-forcing the Password
+
+The default attack style of the ZAP Fuzzer when multiple payload positions are assigned a payload set is the [cluster bomb attack](https://portswigger.net/burp/documentation/desktop/tools/intruder/configure-attack/attack-types). (The ZAP Fuzzer is equivalent to Burp Suite’s Intruder.) A cluster bomb attack iterates through all possible combinations of the payloads. However, we want a [pitchfork attack](https://portswigger.net/burp/documentation/desktop/tools/intruder/configure-attack/attack-types) to run through our username and password lists simultaneously. So, we’re using custom scripts in both of the methods below.
+
+### Method 1: Interleave Wordlists With Valid Credentials
+
+Based on our tests in the previous section, we need to mix our username and password lists with our own credentials, `wiener:peter`. The username list will then look like this:
+
+```
+carlos
+carlos
+wiener
+carlos
+carlos
+wiener
+carlos
+```
+
+The password list will look similar, with “peter” repeated every third line. If you haven’t already, copy the [passwords in the provided wordlist](https://portswigger.net/web-security/authentication/auth-lab-passwords):
+
+```
+123456
+password
+peter
+12345678
+qwerty
+peter
+123456789
+```
+
+I used [a script](https://gist.github.com/Wryhder/041c6d1bed5ab3433c8ea3dd3a49169b) to mix in the valid credentials, but you could use online text tools. After you have your username and passwords lists, create a new Payload Generator script.
+
+#### Step 1: Create a New Payload Generator Script
+
+[Payload generators](https://www.zaproxy.org/docs/desktop/addons/fuzzer/payloads/) in ZAP generate the inputs sent to the target site you’re attacking. There are generators for strings, numbers, and other data types. You can also create custom scripts to generate specific payloads that the default generators don’t cover.
+
+Create a new Payload Generator script by right-clicking the “Payload Generator” type in the *Scripts* tab and hitting *New Script*. Or, use the *New Script* icon in the top-left corner:
```

Comment:
If you've generated the lists there's no need to use a script for the payloads, just use the text files.